cmd/kwil-cli/cmds/common/roundtripper.go

```diff
@@ -54,18 +55,23 @@ func DialClient(ctx context.Context, cmd *cobra.Command, flags uint8, fn RoundTr
 		if needPrivateKey { // only check chain ID if signing something
 			clientConfig.ChainID = conf.ChainID
 		}
-	} else if flags&WithoutPrivateKey == 0 {
+	} else if flags&WithoutPrivateKey == 0 && flags&AuthenticatedCalls == 0 {
+		// private key checks for call messages are done after creating the client
+		// as this requires whether the Kwild node is in private mode or not.
 		return fmt.Errorf("private key not provided")
 	}
 
 	// if not using the gateway, then we can simply create a regular client and return
 	if flags&UsingGateway == 0 {
-		clientConfig.AuthenticateCalls = (flags&AuthenticatedCalls != 0)
 		client, err := client.NewClient(ctx, conf.Provider, &clientConfig)
 		if err != nil {
 			return err
 		}
 
+		if client.PrivateMode() && flags&AuthenticatedCalls != 0 && conf.PrivateKey == nil {
```

Comment:
This isn't going to work. You do:

```go
if flags&UsingGateway == 0 {
// ...
}
```

but the only way the CLI knows it is talking to the gateway is if the `--authenticate` flag was passed. Therefore, you would never be in this condition if the user was trying to authenticate.

I think the larger issue is that the `UsingGateway` flag is no longer applicable / accurate to what the application is doing, because it will always be true if a user passes `--authenticate`. I really think this code should be refactored, because at this point what some of these variables means is entirely innacurate.

Reply:
This might have to turn this PR into a larger fix of consolidating gateway client and regular client. This isn't ideal simply because it is a lot of work, but it might be unavoidable.